astropy/io/fits/tests/test_fitsheader.py

```diff
@@ -0,0 +1,161 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import sys
+
+import numpy as np
+import re
+
+from . import FitsTestCase
+from ..hdu import PrimaryHDU
+from ..scripts import fitsheader
+from ....tests.helper import catch_warnings
+from ....tests.helper import pytest
+from ....utils.exceptions import AstropyDeprecationWarning
+from ....version import version
+ 
+# @pytest.fixture(scope=module , params=['./tests/test0.fits'])
+# def resource_testfile_setup(request):
+#     with open(request.params[0]) as test_fits:
+#         yield test_fits
+
+@pytest.fixture(scope='module')
+def warning_regex():
+    regex = re.compile('^WARNING.*')
+    return regex
+
+@pytest.fixture(scope='module')
+def error_regex():
+    regex = re.compile('^ERROR.*')
+    return regex
+
+class TestFITSheader_script(FitsTestCase):
+
+    def setup_method(self,  method):
+        self.sys_argv_orig = sys.argv
```

Comment:
@pllim `subprocess` does work on windows AFAIK. Also I will remove the `xfail` things. I understand the usage better now. I will add the test for the table formatting and go on with `fitscheck`.